lib/tapioca/dsl/helpers/active_record_column_type_helper.rb

```diff
@@ -148,6 +147,23 @@ def serialized_column_type(column_type)
             "T.untyped"
           end
         end
+
+        sig { params(column_type: T.untyped, getter_type: String).returns(String) }
+        def maybe_nilable_type(column_type, getter_type)
+          return as_nilable_type(getter_type) unless column_type.is_a?(ActiveRecord::Type::Serialized)
+
+          case column_type.coder
+          when ActiveRecord::Coders::YAMLColumn
+            case column_type.coder.object_class
+            when Array.singleton_class, Hash.singleton_class
+              getter_type
+            else
+              as_nilable_type(getter_type)
+            end
+          else
+            as_nilable_type(getter_type)
+          end
```

Comment:
What do you think of this to remove the outer case?

```suggestion
          return as_nilable_type(getter_type) unless column_type.is_a?(ActiveRecord::Type::Serialized) &&
            column_type.coder.is_a?(ActiveRecord::Coders::YAMLColumn)

          case column_type.coder.object_class
          when Array.singleton_class, Hash.singleton_class
            getter_type
          else
            as_nilable_type(getter_type)
          end
```